### PR TITLE
fix: Popover / Tooltip jump when height change

### LINKED ISSUE #44847: 当 Popover 内容变化时，位置会出现闪烁
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-forked-pz4kv9?file=/index.js)

### Steps to reproduce

1. 打开复现链接
2. 鼠标放在最下面的几个列表项的头像上面，会出现Popover
3. 仔细观察Popover的位置，在1s之后，Popover的内容会改变，此时位置会闪烁一下

### What is expected?

1. 相对于触发元素的位置不要发生变化

### What is actually happening?

1. 相对于触发元素的位置发生变化并迅速重新计算，会使页面元素看起来闪烁了一下

| Environment | Info |
| --- | --- |
| antd | 5.9.0 |
| React | 18.2.0 |
| System | MacOS 13.3 (22E252) |
| Browser | Google Chrome 116.0.5845.96 (x86_64) |

---
这可能和 popover 定位方式有关，观察到 popover 定位是通过 inset css 属性实现的，
但是无论 popover 的方向是什么，位置总是由 top 和 left 控制，这就造成了内容尺寸变化时，需要重新计算调整位置，造成视觉看起来位置快速闪烁了一下。

如果针对不同方向的 popover 使用不同的属性控制，应该可以在一定程度上解决此问题，
如 placement top 使用 bottom 和 left 控制，placement bottom 使用 top 和 left 控制。

---

https://github.com/ant-design/ant-design/assets/11692935/880c072d-4f92-4de8-ab38-4810dc70fb94

---

Ref #43187 #42881

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix #44847

### 💡 Background and solution

类 Select 组件做过，Tooltip 用的 `builtinPlacement` 不是同一套，补上

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Fix Tooltip / Popover position jump when content height changed.      |
| 🇨🇳 Chinese |     修复 Tooltip / Popover 在内容高度变化时，位置会闪动的问题。           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 727b0ee</samp>

This pull request adds a new experimental feature for the tooltip component, which makes it more responsive to the layout and viewport. It also improves the code quality and appearance of the tooltip demo files, by using flexbox layout, adjusting button widths, and refactoring style props. The affected files are `components/_util/placements.ts`, `components/tooltip/demo/arrow.tsx`, and `components/tooltip/demo/placement.tsx`.

### <samp>🤖 Generated by Copilot at 727b0ee</samp>

* Enable dynamic inset feature for tooltip component ([link](https://github.com/ant-design/ant-design/pull/44976/files?diff=unified&w=0#diff-070d8296fed37973d7c8a2606b58e5a87b7d40ccb36fcb41d099a51af601730eR159-R161))
* Refactor `placement.tsx` demo file to use flexbox layout ([link](https://github.com/ant-design/ant-design/pull/44976/files?diff=unified&w=0#diff-e536d7a6a9684c924d2d6998d5240ce2a902c1760ca49cd1228ddee2f54cc9c6L6-R82))
* Update `buttonWidth` variable in `arrow.tsx` demo file ([link](https://github.com/ant-design/ant-design/pull/44976/files?diff=unified&w=0#diff-9d8282bfef88783426ca34cf677d55aa7d0823b54db0e0ccb2d256262b0524abL6-R6))
* Add empty line at the beginning of `placements.ts` file for code style consistency ([link](https://github.com/ant-design/ant-design/pull/44976/files?diff=unified&w=0#diff-070d8296fed37973d7c8a2606b58e5a87b7d40ccb36fcb41d099a51af601730eR3))

### CODE REVIEW

**Review by li-jia-nan — APPROVED** [review]

File: components/_util/placements.ts [hunk 1]
```diff
@@ -155,6 +156,9 @@ export default function getPlacements(config: PlacementsConfig) {
     const placementInfo = {
       ...template,
       offset: [0, 0],
+      _experimental: {
```
Comment (afc163) [inline components/_util/placements.ts:159]: 正式版本代码里写 _experimental ，感觉不太好。

Comment (afc163) [reply components/_util/placements.ts:159]: 转正一下？

Comment (zombieJ) [reply components/_util/placements.ts:159]: 感觉是会有更好的名字，但是没想到。所以 `_experimental` 避免开发者用 `@rc-component/trigger` 里的这个属性，没意识到可能会被重构掉。

Comment (afc163) [reply components/_util/placements.ts:159]: 想一个呗，antd 都用了就不能算 _experimental 了。

**Pull request merged**